### PR TITLE
Added the Dune 2000 Guarding voice

### PR DESCRIPTION
**Pull request by Mailaender**
as requested in https://github.com/OpenRA/OpenRA/pull/7147.

### CODE REVIEW

**Comment by Phrohdoh** [comment]
The *Info reference should be readonly.

**Comment by Mailaender** [comment]
Right. Updated.

**Comment by Phrohdoh** [comment]
Looks good to me but I can't test atm.

File: OpenRA.Mods.RA/Guard.cs [hunk 1]
```diff
@@ -18,10 +18,22 @@
 namespace OpenRA.Mods.RA.Traits
 {
 	[Desc("The player can give this unit the order to follow and protect friendly units with the Guardable trait.")]
-	class GuardInfo : TraitInfo<Guard> { }
+	class GuardInfo : ITraitInfo
+	{
+		public readonly string VoicePhrase = "Move";
```
Comment (ScottNZ) [inline OpenRA.Mods.RA/Guard.cs]: This field should be simply called `Voice`. See C4Demolition.cs, Selectable.cs, etc

Comment (Mailaender) [inline OpenRA.Mods.RA/Guard.cs]: Sure thing. Updated.

**Comment by obrakmann** [comment]
:+1:

**Comment by penev92** [comment]
TD, RA and TS show no regression.
Sounds good in D2k, except for the sardaukar, which currently crash the game with
>System.InvalidOperationException was unhandled
  Message=Can't find Guard in voice pool.
  StackTrace:
       at OpenRA.Sound.PlayPredefined(Ruleset ruleset, Player p, Actor voicedUnit, String type, String definition, String variant, Boolean relative, WPos pos, Single volumeModifier, Boolean attenuateVolume) in OpenRA.Game\Sound\Sound.cs:line 331

Also why are the the Fremen and the Saboteurs using the Atreides and Ordos sounds respectively, regardless of what house you are?

File: OpenRA.Mods.RA/Guard.cs [hunk 2]
```diff
@@ -18,10 +18,22 @@
...
 	[Desc("The player can give this unit the order to follow and protect friendly units with the Guardable trait.")]
-	class GuardInfo : TraitInfo<Guard> { }
+	class GuardInfo : ITraitInfo
+	{
+		public readonly string Voice = "Move";
+
+		public object Create(ActorInitializer init) { return new Guard(this); }
+	}
 
 	class Guard : IResolveOrder, IOrderVoice
```
Comment (penev92) [inline OpenRA.Mods.RA/Guard.cs]: Can we rename this to `Guards` ?
`Guard` sounds more like like an activity than a trait.

Comment (Mailaender) [inline OpenRA.Mods.RA/Guard.cs]: Definitely not in this pull request. Make a separate ticket for that.

Comment (penev92) [inline OpenRA.Mods.RA/Guard.cs]: OK.

**Comment by Mailaender** [comment]
Rebased and fixed the crash regarding Sardaukar.

> Also why are the Fremen and the Saboteurs using the Atreides and Ordos sounds respectively, regardless of what house you are?

Probably because they are special units and they did not record for houses that usually can't train them.

**Comment by penev92** [comment]
But the "Guarding" notification is not unit-specific. The ... "House assistant", let's call it, says it. (You know, the one that says "Building" and stuff).

**Comment by penev92** [comment]
OK I don't know how it was in the original game, but if we chose not to use player-House's-voice for these House-specific units, the Sardaukar will need to use the Harkonnen voice.

**Comment by Unit158** [comment]
Sardaukar had their own sound in the original.

**Comment by huwpascoe** [comment]
Can confirm this works :+1:

**Comment by Mailaender** [comment]
Don't merge this yet. I can make the guarding voices for the special units current house specific, too.

**Comment by penev92** [comment]
I think that would feel less strange.

**Comment by penev92** [comment]
> Can confirm this works :+1:

Partially.

**Comment by Mailaender** [comment]
Updated.

**Comment by penev92** [comment]
This now works :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/c3a43712de1a590f53977d7e668de194ebc1d5b9%5E...c3a43712de1a590f53977d7e668de194ebc1d5b9)

**Pull request merged**